packages/block-editor/src/store/private-actions.js

```diff
@@ -131,12 +131,28 @@ export const privateRemoveBlocks =
 			// corresponding to "important" blocks, i.e. blocks that require a
 			// removal prompt.
 			const queue = [ ...clientIds ];
+			let messageType = 'templates';
 			while ( queue.length ) {
 				const clientId = queue.shift();
 				const blockName = select.getBlockName( clientId );
 				if ( rules[ blockName ] ) {
 					blockNamesForPrompt.add( blockName );
 				}
+
+				if ( rules[ 'bindings/core/pattern-overrides' ] ) {
+					const blockAttributes =
+						select.getBlockAttributes( clientId );
+					if (
+						blockAttributes?.metadata?.bindings &&
+						JSON.stringify(
+							blockAttributes.metadata.bindings
+						).includes( 'core/pattern-overrides' )
```

Comment:
because core/patter-overrides is nested under a range of different bindings, eg. content, url, etc. this seemed like the easiest way to check without needed to iterate through them all.

Reply:
Why do have all these block specific logic in place? (not specific to this PR but I feel like we take too much shortcuts and hacks: filters, special cases... we should try to do a bit better and take some time to figure out the right APIs...)